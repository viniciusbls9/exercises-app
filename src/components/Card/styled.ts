import styled from 'styled-components';

export const CardContainer = styled.div`
  width: 300px;
  height: 300px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  padding: 20px;
  margin-right: 50px;
  background-color: #f7f7f7;
  margin-bottom: 20px;
  border-bottom: 2px solid transparent;
  border-right: 2px solid #ccc;
  transition: border-bottom-color 0.3s ease-in-out;

  &:hover {
    border-bottom-color: #ccc;
  }

  @media (max-width: 768px) {
    width: 200px;
    height: 200px;
    margin-right: 20px;
  }
`;

export const Icon = styled.img`
  width: 50px;
  height: 50px;
  margin-bottom: 10px;
`;

export const Text = styled.p`
  font-size: 16px;
  text-align: center;
`;