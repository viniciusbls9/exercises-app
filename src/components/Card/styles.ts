import styled from 'styled-components';

export const CardContainer = styled.div`
  width: 200px;
  height: 200px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 20px;
  margin-bottom: 20px;
  border-bottom: 2px solid transparent;
  transition: border-bottom-color 0.3s ease-in-out;

  &:hover {
    border-bottom-color: #ccc;
  }
`;

export const Icon = styled.img`
  width: 50px;
  height: 50px;
  margin-bottom: 10px;
`;

export const Text = styled.p`
  font-size: 16px;
  text-align: center;
`;