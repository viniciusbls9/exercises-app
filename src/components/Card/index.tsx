import React, { ReactNode } from 'react';
import * as Styles from './styles'

interface CardProps {
  icon: JSX.Element
  text: string
}

const Card = ({ icon, text }: CardProps) => {
  return (
    <Styles.CardContainer>
      {icon}
      <Styles.Text>{text}</Styles.Text>
    </Styles.CardContainer>
  );
};

export default Card;
