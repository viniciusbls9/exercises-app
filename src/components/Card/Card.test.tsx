import { render, screen } from '@testing-library/react';
import Card from './';
import { Activity } from '@phosphor-icons/react';

describe('Card', () => {
  it('renders the icon and text', () => {
    const mockIcon = <Activity alt="Card Icon" />;
    const mockText = 'Card Text';

    const { container } = render(<Card icon={mockIcon} text={mockText} />);

    const textElement = screen.getByText(mockText);

    expect(textElement).toBeInTheDocument();
    expect(container).toMatchSnapshot()
  });
});