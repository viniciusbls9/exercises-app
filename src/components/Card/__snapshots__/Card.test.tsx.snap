// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > renders the icon and text 1`] = `
<div>
  <div
    class="sc-aXZVg iMZrSs"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Card Icon
      </title>
      <path
        d="M240,128a8,8,0,0,1-8,8H204.94l-37.78,75.58A8,8,0,0,1,160,216h-.4a8,8,0,0,1-7.08-5.14L95.35,60.76,63.28,131.31A8,8,0,0,1,56,136H24a8,8,0,0,1,0-16H50.85L88.72,36.69a8,8,0,0,1,14.76.46l57.51,151,31.85-63.71A8,8,0,0,1,200,120h32A8,8,0,0,1,240,128Z"
      />
    </svg>
    <p
      class="sc-eqUAAy homPnk"
    >
      Card Text
    </p>
  </div>
</div>
`;
